src/transports/HTTPTransport.ts

```diff
@@ -27,8 +29,19 @@ class HTTPTransport implements ITransport {
       this.onDataCallbacks.map((cb) => {
         cb(result);
       });
+    }).catch(async (e) => {
+      if (!(e instanceof Error)) {
+        const responseText = await e.text();
```

Comment:
in theory I think there shouldn't be any response text. given the response shouldn't throw an error unless its a non responding server.